format/Schema.fbs

```diff
@@ -298,13 +298,20 @@ enum Endianness:short { Little, Big }
 
 /// ----------------------------------------------------------------------
 /// A Buffer represents a single contiguous memory segment
+
 struct Buffer {
   /// The relative offset into the shared memory page where the bytes for this
   /// buffer starts
   offset: long;
 
-  /// The absolute length (in bytes) of the memory buffer. The memory is found
-  /// from offset (inclusive) to offset + length (non-inclusive).
+  /// The absolute length (in bytes) of the memory buffer. The memory
+  /// is found from offset (inclusive) to offset + length
+  /// (non-inclusive). In the IPC protocol, buffer lengths are always
+  /// padded to a minimum size multiple of 8, but the length here is
+  /// the Buffer size that the writer intends the reader to
+  /// receive. For example, a buffer of size 1 may go on the wire with
+  /// 1 byte plus 7 bytes of padding, though the length here can
+  /// either be 1 or 8
```

Comment:
Are you sure about "either"? In the streaming format, the physical length should be known by the receiver in order to avoid desynchronization (especially if padding > 8)?

Reply:
@pitrou this buffer is within the Message body, which must be received atomically. I can clarify